bridge_adaptivity/module/models.py

```diff
@@ -42,7 +40,26 @@ def _get_engine_driver(engine):
     return driver
 
 
-ENGINES = _discover_engines()
+def _load_cls_from_applicable_module(module_path, mod_name, class_startswith=None, class_endswith=None):
+    """Load class from module."""
+    module = None
+    gp_module = importlib.import_module('{}{}'.format(module_path, mod_name))
+    for attr in inspect.getmembers(gp_module):
+        if class_endswith and attr[0].endswith(class_endswith):
+            module = attr[1]
+        elif class_startswith and attr[0].startswith(class_startswith):
+            module = attr[1]
+    return module
+
+
+ENGINES = _discover_applicable_modules(folder_name='engines', file_startswith='engine_')
+
+GRADING_POLICY_MODULES = _discover_applicable_modules(folder_name='policies', file_startswith='policy_')
+GRADING_POLICY_NAME_TO_CLS = {
+    name: _load_cls_from_applicable_module("module.policies.policy_", name, class_endswith="GradingPolicy")
```

Comment:
I would use `file_name` instead of `name` in _load_cls_from_applicable_module to clarify parameters usage:
```
GRADING_POLICY_NAME_TO_CLS = {
    name: _load_cls_from_applicable_module("module.policies", file_name, class_endswith="GradingPolicy")
    for file_name, name in GRADING_POLICY_MODULES
 }
```

Of course, it is up to you.

Reply:
I would not agree with you. I use this dict in other places where I need to translate policy_name to policy class.